pkg/cmd/pr/checks/checks.go

```diff
@@ -161,6 +176,10 @@ func checksRun(opts *ChecksOptions) error {
 		return err
 	}
 
+	if opts.Exporter != nil {
+		return opts.Exporter.Write(opts.IO, checks)
+	}
```

Comment:
This doesn't seem to do any filtering on `--json` flag. E.g.:

```shell
$ go run cmd/gh/main.go pr checks https://github.com/cli/cli/pull/9079 --json name | jq | head -n 15
[
  {
    "name": "integration-tests (ubuntu-latest)",
    "state": "SUCCESS",
    "startedAt": "2024-05-14T06:02:08Z",
    "completedAt": "2024-05-14T06:02:21Z",
    "link": "https://github.com/cli/cli/actions/runs/9074506128/job/24933385414",
    "bucket": "pass",
    "event": "pull_request",
    "workflow": "Unit and Integration Tests",
    "description": ""
  },
  {
    "name": "integration-tests (windows-latest)",
    "state": "IN_PROGRESS",
```

However, based on https://github.com/cli/cli/blob/4b077daf7e74dbe6bf933f69e220066e3a12e8b4/pkg/cmd/repo/deploy-key/list/list.go#L75-L77, it seems that no extra code is required to do the work. So I'm confused about what's missing 🤔

Reply:
```shell
andyfeller@Andys-MBP:cli/cli ‹pr_checks_json›$ make
go build -trimpath -ldflags "-X github.com/cli/cli/v2/internal/build.Date=2024-05-14 -X github.com/cli/cli/v2/internal/build.Version=v2.49.2-3-g8a1995c9 " -o bin/gh ./cmd/gh
go: downloading google.golang.org/protobuf v1.34.1
go: downloading github.com/sigstore/protobuf-specs v0.3.2


andyfeller@Andys-MBP:cli/cli ‹pr_checks_json›$ ./bin/gh pr checks 9079 --json
Specify one or more comma-separated fields for `--json`:
  bucket
  completedAt
  description
  event
  link
  name
  startedAt
  state
  workflow


andyfeller@Andys-MBP:cli/cli ‹pr_checks_json›$ ./bin/gh pr checks 9079 --json name,state
[
  {
    "name": "CodeQL",
    "state": "SUCCESS",
    "startedAt": "2024-05-14T06:09:38Z",
    "completedAt": "2024-05-14T06:09:42Z",
    "link": "https://github.com/cli/cli/runs/24933625166",
    "bucket": "pass",
    "event": "",
    "workflow": "",
    "description": ""
  },
  {
    "name": "build (windows-latest)",
    "state": "SUCCESS",
    "startedAt": "2024-05-14T06:07:47Z",
    "completedAt": "2024-05-14T06:18:28Z",
    "link": "https://github.com/cli/cli/actions/runs/9074565382/job/24933550081",
    "bucket": "pass",
    "event": "push",
    "workflow": "Unit and Integration Tests",
    "description": ""
  },
  {
    "name": "build (windows-latest)",
    "state": "SUCCESS",
    "startedAt": "2024-05-14T06:07:23Z",
    "completedAt": "2024-05-14T06:15:00Z",
    "link": "https://github.com/cli/cli/actions/runs/9074565588/job/24933550650",
    "bucket": "pass",
    "event": "pull_request",
    "workflow": "Unit and Integration Tests",
    "description": ""
  },
  {
    "name": "integration-tests (windows-latest)",
    "state": "SUCCESS",
    "startedAt": "2024-05-14T06:07:23Z",
    "completedAt": "2024-05-14T06:08:52Z",
    "link": "https://github.com/cli/cli/actions/runs/9074565588/job/24933550798",
    "bucket": "pass",
    "event": "pull_request",
    "workflow": "Unit and Integration Tests",
    "description": ""
  },
  {
    "name": "lint",
    "state": "SUCCESS",
    "startedAt": "2024-05-14T06:07:21Z",
    "completedAt": "2024-05-14T06:08:57Z",
    "link": "https://github.com/cli/cli/actions/runs/9074565587/job/24933550288",
    "bucket": "pass",
    "event": "pull_request",
    "workflow": "Lint",
    "description": ""
  },
  {
    "name": "integration-tests (windows-latest)",
    "state": "SUCCESS",
    "startedAt": "2024-05-14T06:07:20Z",
    "completedAt": "2024-05-14T06:11:59Z",
    "link": "https://github.com/cli/cli/actions/runs/9074565382/job/24933549727",
    "bucket": "pass",
    "event": "push",
    "workflow": "Unit and Integration Tests",
    "description": ""
  },
  {
    "name": "CodeQL-Build",
    "state": "SUCCESS",
    "startedAt": "2024-05-14T06:07:20Z",
    "completedAt": "2024-05-14T06:09:47Z",
    "link": "https://github.com/cli/cli/actions/runs/9074565586/job/24933550178",
    "bucket": "pass",
    "event": "pull_request",
    "workflow": "Code Scanning",
    "description": ""
  },
  {
    "name": "integration-tests (ubuntu-latest)",
    "state": "SUCCESS",
    "startedAt": "2024-05-14T06:07:19Z",
    "completedAt": "2024-05-14T06:07:32Z",
    "link": "https://github.com/cli/cli/actions/runs/9074565588/job/24933550518",
    "bucket": "pass",
    "event": "pull_request",
    "workflow": "Unit and Integration Tests",
    "description": ""
  },
  {
    "name": "integration-tests (macos-latest)",
    "state": "SUCCESS",
    "startedAt": "2024-05-14T06:07:19Z",
    "completedAt": "2024-05-14T06:07:43Z",
    "link": "https://github.com/cli/cli/actions/runs/9074565588/job/24933551059",
    "bucket": "pass",
    "event": "pull_request",
    "workflow": "Unit and Integration Tests",
    "description": ""
  },
  {
    "name": "lint",
    "state": "SUCCESS",
    "startedAt": "2024-05-14T06:07:18Z",
    "completedAt": "2024-05-14T06:08:53Z",
    "link": "https://github.com/cli/cli/actions/runs/9074565383/job/24933549315",
    "bucket": "pass",
    "event": "push",
    "workflow": "Lint",
    "description": ""
  },
  {
    "name": "build (ubuntu-latest)",
    "state": "SUCCESS",
    "startedAt": "2024-05-14T06:07:18Z",
    "completedAt": "2024-05-14T06:10:35Z",
    "link": "https://github.com/cli/cli/actions/runs/9074565588/job/24933550239",
    "bucket": "pass",
    "event": "pull_request",
    "workflow": "Unit and Integration Tests",
    "description": ""
  },
  {
    "name": "build (macos-latest)",
    "state": "SUCCESS",
    "startedAt": "2024-05-14T06:07:18Z",
    "completedAt": "2024-05-14T06:11:14Z",
    "link": "https://github.com/cli/cli/actions/runs/9074565588/job/24933550925",
    "bucket": "pass",
    "event": "pull_request",
    "workflow": "Unit and Integration Tests",
    "description": ""
  },
  {
    "name": "build (ubuntu-latest)",
    "state": "SUCCESS",
    "startedAt": "2024-05-14T06:07:17Z",
    "completedAt": "2024-05-14T06:10:32Z",
    "link": "https://github.com/cli/cli/actions/runs/9074565382/job/24933549552",
    "bucket": "pass",
    "event": "push",
    "workflow": "Unit and Integration Tests",
    "description": ""
  },
  {
    "name": "integration-tests (macos-latest)",
    "state": "SUCCESS",
    "startedAt": "2024-05-14T06:07:17Z",
    "completedAt": "2024-05-14T06:07:44Z",
    "link": "https://github.com/cli/cli/actions/runs/9074565382/job/24933549902",
    "bucket": "pass",
    "event": "push",
    "workflow": "Unit and Integration Tests",
    "description": ""
  },
  {
    "name": "integration-tests (ubuntu-latest)",
    "state": "SUCCESS",
    "startedAt": "2024-05-14T06:07:16Z",
    "completedAt": "2024-05-14T06:07:29Z",
    "link": "https://github.com/cli/cli/actions/runs/9074565382/job/24933549370",
    "bucket": "pass",
    "event": "push",
    "workflow": "Unit and Integration Tests",
    "description": ""
  },
  {
    "name": "build (macos-latest)",
    "state": "SUCCESS",
    "startedAt": "2024-05-14T06:07:16Z",
    "completedAt": "2024-05-14T06:10:44Z",
    "link": "https://github.com/cli/cli/actions/runs/9074565382/job/24933550257",
    "bucket": "pass",
    "event": "push",
    "workflow": "Unit and Integration Tests",
    "description": ""
  }
]
```

Let's take a look as this should only be exporting the fields specified.

Reply:
After pushing up the `ExportData` implementation, this is looking better:

```shell
andyfeller@Andys-MBP:cli/cli ‹pr_checks_json*›$ ./bin/gh pr checks 9079 --json name,state
[
  {
    "name": "CodeQL",
    "state": "SUCCESS"
  },
  {
    "name": "build (windows-latest)",
    "state": "SUCCESS"
  },
  {
    "name": "build (windows-latest)",
    "state": "SUCCESS"
  },
  {
    "name": "integration-tests (windows-latest)",
    "state": "SUCCESS"
  },
  {
    "name": "lint",
    "state": "SUCCESS"
  },
  {
    "name": "integration-tests (windows-latest)",
    "state": "SUCCESS"
  },
  {
    "name": "CodeQL-Build",
    "state": "SUCCESS"
  },
  {
    "name": "integration-tests (ubuntu-latest)",
    "state": "SUCCESS"
  },
  {
    "name": "integration-tests (macos-latest)",
    "state": "SUCCESS"
  },
  {
    "name": "lint",
    "state": "SUCCESS"
  },
  {
    "name": "build (ubuntu-latest)",
    "state": "SUCCESS"
  },
  {
    "name": "build (macos-latest)",
    "state": "SUCCESS"
  },
  {
    "name": "build (ubuntu-latest)",
    "state": "SUCCESS"
  },
  {
    "name": "integration-tests (macos-latest)",
    "state": "SUCCESS"
  },
  {
    "name": "integration-tests (ubuntu-latest)",
    "state": "SUCCESS"
  },
  {
    "name": "build (macos-latest)",
    "state": "SUCCESS"
  }
]
```

Reply:
I concur:
```shell
$ go run cmd/gh/main.go pr checks https://github.com/cli/cli/pull/9079 --json name
[
  {
    "name": "CodeQL"
  },
  {
    "name": "build (windows-latest)"
  },
  {
    "name": "build (windows-latest)"
   },
   ...
]
```

Reply:
This can be closed, thanks @andyfeller for finding the fix!